File: p_027_1_getintentaction/src/main/java/com/startandroid/p_027_1_getintentaction/MainActivity.java
package com.startandroid.p_027_1_getintentaction;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnT = findViewById(R.id.btnT);
        btnT.setOnClickListener(this);

        Button btnD = findViewById(R.id.btnD);
        btnD.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent intent;
        switch (v.getId()) {
            case R.id.btnD:
                intent = new Intent("com.startandroid.intent.action.showdate");
                startActivity(intent);
                break;
            case R.id.btnT:
                intent = new Intent("com.startandroid.intent.action.showtime");
                startActivity(intent);
                break;

        }

    }
}

File: k_092_servise/src/main/java/com/startandroid/k_092_servise/MainActivity.kt
package com.startandroid.k_092_servise

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity(R.layout.activity_main) {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //startService(Intent(this, MyService::class.java))

        val start = findViewById<Button>(R.id.btnStart)
        val stop = findViewById<Button>(R.id.btnStop)

        start.setOnClickListener {
            startService(Intent(this, MyService::class.java))
        }

        stop.setOnClickListener {
            stopService(Intent(this, MyService::class.java))
        }
    }
}